storagenode/storagenodedb/database.go

```diff
@@ -115,108 +117,83 @@ func New(log *zap.Logger, config Config) (*DB, error) {
-		ndb:    teststore.New(),
-		adb:    teststore.New(),
-
-		// Initialize databases. Currently shares one info.db database file but
-		// in the future these will initialize their own database connections.
-		versionsDB:        newVersionsDB(versionsDB, versionsPath),
-		v0PieceInfoDB:     newV0PieceInfoDB(versionsDB, versionsPath),
-		bandwidthDB:       newBandwidthDB(versionsDB, versionsPath),
-		ordersDB:          newOrdersDB(versionsDB, versionsPath),
-		pieceExpirationDB: newPieceExpirationDB(versionsDB, versionsPath),
-		pieceSpaceUsedDB:  newPieceSpaceUsedDB(versionsDB, versionsPath),
-		reputationDB:      newReputationDB(versionsDB, versionsPath),
-		storageUsageDB:    newStorageusageDB(versionsDB, versionsPath),
-		usedSerialsDB:     newUsedSerialsDB(versionsDB, versionsPath),
-		satellitesDB:      newSatellitesDB(versionsDB, versionsPath),
+		return errs.Combine(err, db.closeDatabases())
 	}
-	return db, nil
+	db.legacyInfoDB.Configure(legacyInfoDB)
+	db.bandwidthDB.Configure(legacyInfoDB)
+	db.ordersDB.Configure(legacyInfoDB)
+	db.pieceExpirationDB.Configure(legacyInfoDB)
+	db.v0PieceInfoDB.Configure(legacyInfoDB)
+	db.pieceSpaceUsedDB.Configure(legacyInfoDB)
+	db.reputationDB.Configure(legacyInfoDB)
+	db.storageUsageDB.Configure(legacyInfoDB)
+	db.usedSerialsDB.Configure(legacyInfoDB)
+	db.satellitesDB.Configure(legacyInfoDB)
+	return nil
 }
 
 // openDatabase opens or creates a database at the specified path.
-func openDatabase(path string) (*sql.DB, error) {
+func (db *DB) openDatabase(dbName string) (*sql.DB, error) {
+	path := filepath.Join(db.dbDirectory, db.FilenameFromDBName(dbName))
 	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
 		return nil, err
 	}
 
-	db, err := sql.Open("sqlite3", "file:"+path+"?_journal=WAL&_busy_timeout=10000")
+	sqlDB, err := sql.Open("sqlite3", "file:"+path+"?_journal=WAL&_busy_timeout=10000")
 	if err != nil {
 		return nil, ErrDatabase.Wrap(err)
 	}
 
-	dbutil.Configure(db, mon)
-	return db, nil
-}
-
-// // openTestDatabase creates an in memory database.
-func openTestDatabase() (*sql.DB, error) {
-	// create memory DB with a shared cache and a unique name to avoid collisions
-	db, err := sql.Open("sqlite3", fmt.Sprintf("file:memdb%d?mode=memory&cache=shared", rand.Int63()))
-	if err != nil {
-		return nil, ErrDatabase.Wrap(err)
-	}
+	// This isn't safe for concurrent access but we don't currently access this map concurrently.
+	// If we do in the future it needs some protection.
+	db.sqlDatabases[dbName] = sqlDB
 
-	// Set max idle and max open to 1 to control concurrent access to the memory DB
-	// Setting max open higher than 1 results in table locked errors
-	db.SetMaxIdleConns(1)
-	db.SetMaxOpenConns(1)
-	db.SetConnMaxLifetime(-1)
+	dbutil.Configure(sqlDB, mon)
 
-	mon.Chain("db_stats", monkit.StatSourceFunc(
-		func(cb func(name string, val float64)) {
-			monkit.StatSourceFromStruct(db.Stats()).Stats(cb)
-		}))
+	db.log.Debug(fmt.Sprintf("opened database %s", dbName))
```

Comment:
this would add a lot of noise to storj-sim when starting/shutting down

Reply:
I feel this is important for debugging both storj-sim and even more important for production. That's what `DEBUG` flagged logs are for, aren't they? Especially since you want the ability in the future to locate databases on a separate drive, that may increase more failure scenarios that we will need to troubleshoot and so will our support team.

Reply:
In that case, you would also want to log the path. For production, debug levels are disabled so they wouldn't be of any significant help.

Reply:
Also, wouldn't the error come from some other place instead?